components/scream/src/physics/spa/spa_functions_impl.hpp

```diff
@@ -90,195 +90,187 @@ void SPAFunctions<S,D>
 ::spa_main(
   const SPATimeState& time_state,
   const view_2d<const Spack>& p_tgt,
-  const SPAData&   data_beg,
-  const SPAData&   data_end,
-  const SPAOutput& data_out,
-  Int ncols_tgt,
-  Int nlevs_tgt,
-  Int nswbands,
-  Int nlwbands)
+  const view_2d<      Spack>& p_src,
+  const SPAInput&   data_beg,
+  const SPAInput&   data_end,
+  const SPAInput&   data_tmp,
+  const SPAOutput&  data_out)
+{
+  // Number of bands must match.
+  EKAT_REQUIRE(data_end.data.nswbands==data_beg.data.nswbands);
+  EKAT_REQUIRE(data_end.data.nlwbands==data_beg.data.nlwbands);
+
+  // For now we require input and output to have same geometric sizes
```

Comment:
Minor, but you probably mean "beginning and ending" data instead of "input and output"

Reply:
Actually, I mixed up things. I need
- beg/end _must_ have same geo/bands dims (meaning spa data has same grid for all months), otherwise it's a mess.
- input/output to have same ncols, since we don't do horiz interp in spa_main (for now)
- input/output must have same bands dims (i don't even know why we wouldn't, or how we would handle that).

I'll fix the checks/comments.